tests/utils/print_alp_containers.hpp

```diff
@@ -54,7 +54,7 @@ void print_matrix( std::string name, const MatrixType &A ) {
 					std::cout <<  ",";
 				}
 			} else {
-				std::cout << std::setprecision( 0 ) << "\t" << ".";
+				std::cout << std::setprecision( 0 ) << "\t" << "0, ";
```

Comment:
Is this needed?

Reply:
Since print_matrix is normally used in debug, it seems to be more convenient to have exact values when you copy matrix to eg. python. If there is a good reason I can leave it as `.`.

Reply:
good point. Maybe we could overload with a value of type `T` to print as zero (in which case it uses a numpy-like format) and if not provided a dot.